src/test/java/api/requests/RequestsAPILoanRenewalTests.java

```diff
@@ -65,6 +73,108 @@ public void allowRenewalLoanByBarcodeWhenFirstRequestInQueueIsHold() throws Inte
 
     Response response = loansFixture.attemptRenewal(200, smallAngryPlanet, rebecca);
     assertThat(response.getJson().getString("action"), is("renewed"));
+    // Assert no validation issues, so the renewal is allowed
+    assertThat(response.getJson().getJsonArray("errors"), nullValue());
+
+    // We use SYSTEM_DATE as base for due date renew + 3 weeks
+    // Set mls and seconds to 0 as we're mostly interesting in weeks
+    final DateTime expectedDueDate = DateTime.now(DateTimeZone.UTC)
+      .plusWeeks(3)
+      .withSecondOfMinute(0)
+      .withMillisOfSecond(0);
+    final DateTime actualDueDate = new DateTime(response.getJson()
+      .getInstant("dueDate").toEpochMilli())
+      .withZone(DateTimeZone.UTC)
+      .withSecondOfMinute(0)
+      .withMillisOfSecond(0);
+    assertThat(actualDueDate, is(expectedDueDate));
```

Comment:
Please, check out the following matcher: `TextDateTimeMatcher#isEquivalentTo`. Seems like it can be applies here.

Reply:
@dmytrotur works for me, thanks.